components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -70,6 +70,28 @@ pub fn ltrim(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
     }))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn left(lhs: &Option<Bytes>, rhs: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(match (lhs, rhs) {
+        (Some(lhs), Some(rhs)) => {
+            if *rhs <= 0 {
+                Some(Vec::new())
+            } else {
```

Comment:
BTW, maybe it's better to check if the length of `lhs` is less than `rhs`?

Reply:
Hi, I think it's not necessary to check the length of `lhs` because the `iterator ` do the thing right already.
I'm newbie here so maybe I was wrong. 

... The truth is that I am trying store chars list with a variable and check length of `lhs`, the compiler blocked me.

Reply:
How about something like:
```
if *rhs <= 0 {
    return Ok(Some(Vec::new()))
}
             
let s = std::str::from_utf8(&*lhs).unwrap();
if s.chars().count() > *rhs as usize {
    Some(s.chars().take(*rhs as usize).collect::<String>().into_bytes())
} else {
    Some(s.to_string().into_bytes())
}
```

Reply:
Thanks